cg/cli/workflow/mip_dna/base.py

```diff
@@ -272,33 +261,117 @@ def run(
         raise click.Abort
 
 
-@mip_dna.command()
+@mip_dna.command("resolve-compression")
+@ARGUMENT_CASE_ID
 @OPTION_DRY
 @click.pass_context
-def start(context: click.Context, dry_run: bool = False):
-    """Start all cases that are ready for analysis"""
+def resolve_compression(context: click.Context, case_id: str, dry_run: bool):
+    """Handles cases where decompression is needed before starting analysis"""
     dna_api = context.obj["dna_api"]
-    exit_code = EXIT_SUCCESS
+    prepare_fastq_api = context.obj["prepare_fastq_api"]
+    case_obj = dna_api.db.family(case_id)
+    if not case_obj:
+        LOG.error("Case %s does not exist in Status-DB", case_id)
+        raise click.Abort
+
+    decompression_needed = prepare_fastq_api.is_spring_decompression_needed(case_obj.internal_id)
+    if decompression_needed:
+        decompression_all_samples_started = prepare_fastq_api.start_spring_decompression(
+            case_obj.internal_id, dry_run
+        )
+        if decompression_all_samples_started:
+            LOG.info(
+                f"The analysis for {case_obj.internal_id} could not start, started decompression instead"
+            )
+        else:
+            LOG.warning(
+                "Neither the analysis, nor the decompression needed for every sample, could start for %s",
+                case_obj.internal_id,
+            )
+        raise DecompressionNeededError("Workflow interrupted: decompression is not finished")
+
+    LOG.info("Linking fastq files in housekeeper for case %s", case_id)
+    prepare_fastq_api.check_fastq_links(case_obj.internal_id)
+
+    if prepare_fastq_api.is_spring_decompression_needed(case_obj.internal_id):
```

Comment:
This test was done above as well, I don;t quite understand what's happening here?

Reply:
@Mropat what do you say? I think this was added long time ago to handle the situation that the loop could take time, and then it checked once again so automated compression had not started which would mess everything up. But I am not sure we need to check for this anymore.

Reply:
So, first it checks if spring decompression is needed. This will be a "yes" if housekeeper has spring files and not fastq files.
Second, it will check if fastq files exist and can be added to Housekeeper, and if so, does that for the files where compression finished. This may or may not be done for all samples in the case (!)
Next, it check again for that case the same question as first time. If some samples in case still have spring but not fastq, it should wait a bit more, so the program stops there.
Next cycle, it will ask again, and so on.

Reply:
Thanks for clarification. I think this should stay, so there is an extra check after files are added to housekeeper, perhaps the text "Workflow interrupted: decompression is not finished" should change and explain that all fastq files was not successfully added to housekeeper?